_posts/2019-04-05-omakase-typescript.markdown

```diff
@@ -0,0 +1,179 @@
+code. Here's an example using JavaScript:
+
+```js
+var name = "Danger"
+console.log("Hello, " + danger)
+```
+
+The first line makes a variable (a kind of box you can keep things in) and then the second line outputs text to the
+console (think DOS, or the terminal) `"Hello, Danger"`. JavaScript is designed to work as a scripting language,
+which means the code starts at the top of the file and then goes through line by line. To provide some contrast,
+here is the [same behavior](https://repl.it/repls/VioletredGlisteningInfo) in Java, which is built with different
+language constraints:
+
+```java
+class Main {
+  public static void main(String[] args) {
+    String name = "Danger";
+    System.out.println("Hello, " + name);
+  }
+}
+```
+
+> Note: if you find the naming of Java and JavaScript confusing (it is, they are two completely separate
+> programming languages, no link at all.) JavaScript was named that way because Java was looking to be really the
+> next hot language (it did turn out that way for a decade or two, but now JavaScript is usually the first language
+> people have heard of.)
+
+Aside from having a lot more lines, the Java version comes with a lot of words that aren't necessarily about
+telling the computer exactly what to do, e.g. `class Main {`, `public static void main(String[] args) {`, `}` and
+`}` again. It also has semi-colons at the end of some lines. Java is aimed at building different things, and these
+extra bits of code make sense within the constraints of building a Java app.
+
+To get to my main point though, there is one standout line I'd like us to compare:
+
+```
+// JavaScript
+var name = "Danger"
+// Java
+String name = "Danger";
+```
+
+Both of these lines declare variables called `name` which contain the value `"Danger"`.
+
+In JavaScript you use the abbreviation `var` to declare a variable. Meanwhile, in Java you need to say _what kind
+of data_ the variable contains. In this case the variable contains a `String`. (A string is a programming term for
+a collection of characters. They `"look like this"`. This [5m video](https://www.youtube.com/watch?v=czTWbdwbt7E)
+is a good primer if you want to learn more.)
+
+Both of these variables contain a string, but the difference is that in Java the variable can _only ever contain a
+\_string_, because that's what we said when we created the variable. In JS the variable can change to be
+_anything_, like a number, or a list of dates.
+
+There are many values which can go into a variable, and `var` accepts all of them in JS. In Java though, we have to
+say upfront what can be inside that variable, which is only strings, so:
+
+```js
+// Before in JS
+var name = "Danger"
+// Also OK
+var name = 1
+var name = false
+var name = ["2018-02-03", "2019-01-12"]
+
+// Before in Java
+String name = "Danger";
+// Not OK, the code wouldn't be accepted by Java
+String name = 1;
+String name = false
+String name = {"2018-02-03", "2019-01-12"};
```

Comment:
I don't think `{"2018-02-03", "2019-01-12"}` is a valid java expression. Did you mean `new String[]{"2018-02-03", "2019-01-12"}`? Or just removing noise to help beginners?

Reply:
I threw it into a Java VM and it errored like I thought it would - guess I didn't know how to make it actually succeed the assignment - switching to the working version